src/QsCompiler/Compiler/Templates/CppInterop.cs

```diff
@@ -0,0 +1,168 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.Quantum.QsCompiler.BondSchemas.EntryPoint;
+
+namespace Microsoft.Quantum.QsCompiler.Templates
+{
+    public class ArgumentCpp : Argument
+    {
+        public ArgumentCpp(Argument argument)
+        {
+            this.Name = argument.Name;
+            this.Position = argument.Position;
+            this.Type = argument.Type;
+            this.ArrayType = argument.ArrayType;
+            this.Values = argument.Values.ToList();
+        }
+
+        public string CppType()
+        {
+            return this.Type switch
+            {
+                DataType.BoolType => "char",
+                DataType.IntegerType => "int64_t",
+                DataType.DoubleType => "double",
+                DataType.PauliType => "char",
+                DataType.RangeType => "InteropRange*",
+                DataType.ResultType => "char",
+                DataType.StringType => "const char*",
+                DataType.ArrayType => "InteropArray *",
+            };
+        }
+
+        public string CliOptionString()
+        {
+            if (this.Name.Length == 1)
+            {
+                return $"-{this.Name}";
+            }
+            else
+            {
+                return $"--{this.Name}";
+            }
+        }
+
+        public string? CliDescription()
```

Comment:
I am curious about why this method returns a nullable string rather than just a string.

Reply:
yep good call doesn't need to be, fixed